Task5.c

```diff
@@ -0,0 +1,77 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+char* getString()
+{
+    int strLength = 0;
+    int strCapacity = 1; // place for '\0' 
+    char* string = malloc(sizeof(char));
+    char scannedChar = getchar();
+    while (scannedChar != '\n')
+    {
+        string[strLength] = scannedChar;
+        strLength++;
+        if (strLength >= strCapacity)
+        {
+            strCapacity *= 2;
+            string = realloc(string, strCapacity * sizeof(char));
+        }
+        scannedChar = getchar();
+    }
+    string[strLength] = '\0';
+    return string;
+}
+
+int bracketsType(char bracket1, int bracket2)
+{
+    if (bracket1 == '(' && bracket2 == ')')
+        return 1;
+    if (bracket1 == '{' && bracket2 == '}')
+        return 1;
+    if (bracket1 == '[' && bracket2 == ']')
+        return 1;
+    if (bracket1 == '<' && bracket2 == '>')
+        return 1;
+    return 0;
+}
+
+int main()
+{
+    printf("Enter string to check balance of brackets: ");
+    char* string = getString();
+    int strLength = strlen(string);
+    char* scannedBrackets = malloc((strLength + 1) * sizeof(char));
```

Comment:
```suggestion
    char* stackOfBrackets = malloc((strLength + 1) * sizeof(char));
```
Этот массив хранит не все отсканированные скобки, а стек ещё не сопоставленных скобок. Имя должно это отражать